openrewrite/test/javascript/parser/interface.test.ts

```diff
@@ -0,0 +1,297 @@
+                  readonly name?: string
+              }
+          `)
+        );
+    });
+
+    test('interface with properties, methods and comments', () => {
+        rewriteRun(
+          //language=typescript
+          typeScript(`
+              interface Person {
+                  /*a*/ greet(/*1*/name/*2*/: /*3*/string/*4*/, /*5*/ surname: string/*6*/): /*b*/ void /*c*/
+                  name /*d*/ : string
+                  name(/*11*/name/*22*/: /*33*/string/*44*/): string;
+                  age: /*e*/ number /*f*/
+                  age(): number;
+              }
+          `)
+        );
+    });
+
+    test('interface with function type', () => {
+        rewriteRun(
+          //language=typescript
+          typeScript(`
+              interface Add {
+                  add(): (x: number, y: number) => number;
+              }
+          `)
+        );
+    });
+
+    test('interface with function type and zero param', () => {
+        rewriteRun(
+          //language=typescript
+          typeScript(`
+              interface Add {
+                  produce(): () => number;
+              }
+          `)
+        );
+    });
+
+    test('interface with function type and several return types', () => {
+        rewriteRun(
+          //language=typescript
+          typeScript(`
+              interface Add {
+                  consume(): () => number /*a*/ | /*b*/ void;
+              }
+          `)
+        );
+    });
+
+    test('interface with function type and comments', () => {
+        rewriteRun(
+          //language=typescript
+          typeScript(`
+              interface Add {
+                  /*a*/ add/*b*/(/*c*/)/*d*/ : /*e*/ (/*f*/ x/*g */:/*h*/ number /*i*/, /*j*/y /*k*/: /*l*/ number/*m*/)/*n*/ => /*o*/ number /*p*/;
+              }
+          `)
+        );
+    });
+
+    test.skip('interface with call signature', () => {
+        rewriteRun(
+          //language=typescript
+          typeScript(`
+              interface Add {
+                  greet: (name: string) => string;
+                  (x: number, y: number): number;
+              }
+            `)
+        );
+    });
+
+    test.skip('interface with indexable type', () => {
+        rewriteRun(
+          //language=typescript
+          typeScript(`
+              interface Add {
+                  [index: number]: string
+              }
+            `)
+        );
+    });
+
+    test.skip('interface with hybrid types', () => {
+        rewriteRun(
+          //language=typescript
+          typeScript(`
+              interface Counter {
+                  (start: number): string;   // Call signature
+                  interval: number;          // Property
+                  reset(): void;             // Method
+                  [index: number]: string    // Indexable
+                  add(): (x: number, y: number) => number; //Function signature
+              }
+          `)
+        );
+    });
```

Comment:
can you also try to adding the property signature with `get` and `set` keywords so this is getter setter signatures?

Reply:
There should be additionally implemented:

```
visitGetAccessor(node: ts.GetAccessorDeclaration) {
    return this.visitUnknown(node);
}

visitSetAccessor(node: ts.SetAccessorDeclaration) {
    return this.visitUnknown(node);
}
```